### ISSUE TITLE
Dap-cpptools never starts its debuggee on Windows 10

### ISSUE THREAD

**Issue by flotsamarch** [emacs-lsp/dap-mode#666]
I've been trying to setup dap-cpptools on my Doom Emacs running on **Windows 10**, but can't get it working. 

I installed cpptools with `dap-cpptools-setup` and then in debug template I added path to the executable to `:program` setting value. When I start debug session with `dap-debug` i get the following message in `*Warnings*` buffer:
`Warning (emacs): Initialize request failed: Unable to start debugging. Launch options string provided by the project system is invalid. Unable to determine path to debugger.  Please specify the "MIDebuggerPath" option.`

After specifying path to the debugger as prompted with a value copied from `dap-cpptools-debug-program` variable the debug sessions starts, as indicated by my modeline changing color. But instead of `dap-show-many-windows` only `*cpptools::Run Configuration<X-number-here> - cppdbg: ProgramName.exe*` empty buffer gets opened and nothing happens after.
My debuggee program never starts. 

In the list of buffers i also see _X-number_ of `*cpptools::Run Configuration<1...X> stderr*` which contain the following text:
`waiting for v8 protocol on stdin/stdout
Process cpptools::Run Configuration<1...X-1> stderr finished`
And the same amount of emtpy `*cpptools::Run Configuration<1...X> out*` buffers.

The last of such stderr buffers **doesn't have** the `stderr finished` line. A quick google led me to [this issue](https://github.com/mfussenegger/nvim-dap/issues/512) in nvim-dap repo that suggests this is due to improper configuration on Windows. I have no idea if it's the same issue or how to translate this to elisp config to test.

`dap-stop-thread` does nothing. `dap-continue` is unable to find stopped thread. `dap-switch-thread` does nothing as far as I can tell.

Here's my dap-mode config
```
;; Configure debugger
(use-package! dap-mode
  :defer t
  :config
  (setq! dap-auto-configure-features
         '(sessions locals breakpoints expressions tooltip))
  (remove-hook 'dap-ui-mode-hook #'dap-ui-controls-mode)
  (dap-ui-controls-mode 0)
  (require 'dap-cpptools)
  (dap-auto-configure-mode t)
  (setq dap-lldb-debug-program '("lldb-vscode")) ;; This is my lldb-vscode (not codelldb) setup which currently works but i'm not happy with.
  (setq dap-lldb-debugged-program-function
        (lambda () (read-file-name "Select file to debug."))))
```
**Dap-mode is unpinned** in my package.el, so it's the latest verion. `dap-cpptools-extension-version` variable holds value "1.11.5"

I'd also like to know how to configure dap-codelldb. Couldn't find any guide. Maybe it works while this issue is getting fixed.

**Comment by yyoncho** [comment]
Here you can see sample configuration https://github.com/emacs-lsp/dap-mode/issues/6#issuecomment-800750477 (json)

You can use M-x dap-debug-edit-template to create elisp template with that setting. I guess it can work if you put it on the path as well with the expected name. You may refer to cpptools docs in order to find the correct name.

**Comment by flotsamarch** [comment]
Turns out this is misconfiguration is my fault because I wasn't aware that OpenDebugAD7 is not an actual debugger. Actually its purpose is to communicate to MI debuggers(lldb/gdb) via DAP. 

My assumption was that since cpptools contains Visual Studio Debugger and allows to use it inside VSCode – it is going to somehow allow me to do the same with dap-cpptools inside Emacs. Now I know this is not the case. Maybe this should be mentioned in docs but idk maybe it is just me being stupid.

So when I tried to make a debug template I provided a path to OpenDebugAD, which caused OpenDebugAD to recursively start itself a couple of times, expecting that it launches a debugger and that the debugger will launch the program to debug. The program was never launched because the actual debugger has never started.

Sorry for wasting creator's time due to my ignorance.
